### DISCUSSION TITLE
Is there a better way to model a variant ?

### DISCUSSION THREAD

**Discussion by binajmen** [prisma/prisma#12185] [Q&A] [answered]
Hi,

I have a model `MProduct`. A product can have different prices. In order to find the siblings/variants of a product, I've created the model `MVariant`: two products that are related would have the same `variantId` value. Here is a short version of my model:

```prisma
model MVariant {
  id       String     @id @default(uuid()) @db.Uuid
  products MProduct[]
}

model MProduct {
  id    String @id @default(uuid()) @db.Uuid
  name  String
  price Int

  variant   MVariant? @relation(fields: [variantId], references: [id])
  variantId String?   @db.Uuid
}
```

Is there a better/shorter way to allow that? I have the feeling the `MVariant` model could be avoided, but I'm not sure how (perhaps this is the correct way to implement this?).

The end goal is to be able to retrieve the variants of a product when they exist:
```ts
  const product = await prisma.product.findUnique({
    where: { id: '...' },
    include: {
      variant: { include: { products: true } },
    },
  });
```
would return
```json
{
  "id": "1b97d68a-e0c8-4099-aafe-d670fb284f15",
  "name": "La Chouffe"
  "price": 450,
  "variant": {
    "id": "1b97d68a-e0c8-4099-aafe-d670fb284f1f",
    "products": [
      {
        "id": "1b97d68a-e0c8-4099-aafe-d670fb284f15",
        "name": "La Chouffe"
        "price": 450,
        "variantId": "1b97d68a-e0c8-4099-aafe-d670fb284f1f",
      },
      {
        "id": "1b97d68a-e0c8-4099-aafe-d670fb284f16",
        "name": "La Chouffe"
        "price": 850,
        "variantId": "1b97d68a-e0c8-4099-aafe-d670fb284f1f",
      }
    ]
  }
}
```

Any help would be appreciated 🙏

**Comment by binajmen** [comment, marked as answer]
Re-reading this doc section https://www.prisma.io/docs/concepts/components/prisma-schema/relations/self-relations#one-to-many-self-relations, I ended up with this model:

```prisma
model MProduct {
  id    String @id @default(uuid()) @db.Uuid
  name  String
  price Int

  parentId String?    @db.Uuid
  parent   MProduct?  @relation("Siblings", fields: [parentId], references: [id])
  siblings MProduct[] @relation("Siblings")
}
```
```ts
  const res = await prisma.mProduct.findUnique({
    where: { id: 'd2a7e017-e924-4ca2-905f-3e919eac2eeb' },
    include: { variants: true },
  });
```
```json
{
  "id": "d2a7e017-e924-4ca2-905f-3e919eac2eeb",
  "name": "Beer",
  "price": 0,
  "parentId": null,
  "variants": [
    {
      "id": "b151f8db-82ef-42bf-a8ec-88ad65f38f9c",
      "name": "Beer 25cl",
      "price": 100,
      "parentId": "d2a7e017-e924-4ca2-905f-3e919eac2eeb"
    },
    {
      "id": "99d5b7af-95d9-44b2-bc23-5f788c76f126",
      "name": "Beer 50cl",
      "price": 200,
      "parentId": "d2a7e017-e924-4ca2-905f-3e919eac2eeb"
    }
  ]
}
```

Still open to suggestions, but I feel this is already better.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
